diff --git a/builtin/builtin.lua b/builtin/builtin.lua
index 9b2bed3..bb38b9a 100644
--- a/builtin/builtin.lua
+++ b/builtin/builtin.lua
@@ -359,6 +359,13 @@ function minetest.node_dig(pos, node, digger)
 	end
 end
 
+-- This is used to allow mods to redefine minetest.item_place and so on
+local function redef_wrapper(table, name)
+	return function(...)
+		return table[name](...)
+	end
+end
+
 --
 -- Item definition defaults
 --
@@ -378,12 +385,12 @@ minetest.nodedef_default = {
 	tool_capabilities = nil,
 
 	-- Interaction callbacks
-	on_place = minetest.item_place,
-	on_drop = minetest.item_drop,
+	on_place = redef_wrapper(minetest, 'item_place'), -- minetest.item_place
+	on_drop = redef_wrapper(minetest, 'item_drop'), -- minetest.item_drop
 	on_use = nil,
 
-	on_punch = minetest.node_punch,
-	on_dig = minetest.node_dig,
+	on_punch = redef_wrapper(minetest, 'node_punch'), -- minetest.node_punch
+	on_dig = redef_wrapper(minetest, 'node_dig'), -- minetest.node_dig
 
 	-- Node properties
 	drawtype = "normal",
@@ -429,8 +436,8 @@ minetest.craftitemdef_default = {
 	tool_capabilities = nil,
 
 	-- Interaction callbacks
-	on_place = minetest.item_place,
-	on_drop = minetest.item_drop,
+	on_place = redef_wrapper(minetest, 'item_place'), -- minetest.item_place
+	on_drop = redef_wrapper(minetest, 'item_drop'), -- minetest.item_drop
 	on_use = nil,
 }
 
@@ -447,8 +454,8 @@ minetest.tooldef_default = {
 	tool_capabilities = nil,
 
 	-- Interaction callbacks
-	on_place = minetest.item_place,
-	on_drop = minetest.item_drop,
+	on_place = redef_wrapper(minetest, 'item_place'), -- minetest.item_place
+	on_drop = redef_wrapper(minetest, 'item_drop'), -- minetest.item_drop
 	on_use = nil,
 }
 
